pkg/apis/agones/v1/gameserver.go

```diff
@@ -461,25 +495,44 @@ func (gss *GameServerSpec) Validate(devAddress string) ([]metav1.StatusCause, bo
 				}
 			}
 		}
-		for _, c := range gss.Template.Spec.Containers {
-			validationErrors := validateResources(c)
-			for _, err := range validationErrors {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Field:   "container",
-					Message: err.Error(),
-				})
-			}
-		}
 	}
-	objMetaCauses := validateObjectMeta(&gss.Template.ObjectMeta)
-	if len(objMetaCauses) > 0 {
-		causes = append(causes, objMetaCauses...)
+
+	podTemplateSpecCauses := ValidatePodTemplateSpec(&gss.Template)
+	if len(podTemplateSpecCauses) > 0 {
+		causes = append(causes, podTemplateSpecCauses...)
 	}
 
 	return causes, len(causes) == 0
 }
 
+// ValidatePodTemplateSpec validates the PodTemplateSpec of a GameServer
+func ValidatePodTemplateSpec(spec *corev1.PodTemplateSpec) []metav1.StatusCause {
+	causes := []metav1.StatusCause{}
+
+	// Before calling the validation, the pod template must be converted to the Core version
+	outPodTemplateSpec := core.PodTemplateSpec{}
+	if err := k8s_api_v1.Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec(spec, &outPodTemplateSpec, nil); err != nil {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeUnexpectedServerResponse, // TODO: Confirm the type to be returned
+			Field:   "template",
+			Message: err.Error(),
+		})
+
+		return causes
+	}
+
+	errList := k8s_validation.ValidatePodTemplateSpec(&outPodTemplateSpec, field.NewPath("spec"))
+	for _, err := range errList {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseType(err.Type),
```

Comment:
This should probably be an official type, I would probably use CauseTypeFieldValueInvalid here as well.

Reply:
The problem is that it may pass back the wrong information. There will be situation where the err.Type will not be about invalid values. Instead it will be related to required fields. In this case we are justing sending back to the caller what K8S is returning from the validation process. Does that make sense?

Reply:
Aah, I see - I didn't pick up on the fact it was bringing through K8s types. Then yes, this makes total sense :+1: